src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/nonassociations/NonAssociationsMigrationService.kt

```diff
@@ -65,9 +66,22 @@ class NonAssociationsMigrationService(
   }
 
   override suspend fun migrateNomisEntity(context: MigrationContext<NonAssociationIdResponse>) {
-    log.info("attempting to migrate ${context.body}")
-    val firstOffenderNo = context.body.offenderNo1
-    val secondOffenderNo = context.body.offenderNo2
+    val nonAssociation = context.body
+    log.info("attempting to migrate $nonAssociation")
+    val firstOffenderNo = nonAssociation.offenderNo1
+    val secondOffenderNo = nonAssociation.offenderNo2
+
+    if (nonAssociation.isNotPrimaryNonAssociation()) {
```

Comment:
Bomb straight out if this is not a non-association we're interested in.

Reply:
might look better to do a 
```kotlin
with (context.body) {
   log.info("attempting to migrate $this")
   ...
```
and you won't need to define any of the 3 variables